discord/client.py

```diff
@@ -251,6 +251,20 @@ def __init__(
             VoiceClient.warn_nacl = False
             _log.warning("PyNaCl is not installed, voice will NOT be supported")
 
+    async def __aenter__(self) -> Client:
+        loop = asyncio.get_running_loop()
+        self.loop = loop
+        self.http.loop = loop
+        self._connection.loop = loop
+
+        self._ready = asyncio.Event()
+
+        return self
+
+    async def __aexit__(self) -> None:
```

Comment:
I though `__aexit__` had extra parameters for exc info?

Reply:
Thats true. I skipped them because I thought there is no need.
How about
```suggestion
    async def __aexit__(self, exc_t, exc_v, exc_tb) -> None:
```
?

Reply:
I'd add some typing first (iirc something like `_exc_t: Type[Exception] | None, _exc_v: Exception | None, _exc_tb: str | None`) and private them since they aren't used

Reply:
Done. I setteled for `BaseException` and `TracebackType` since (IIUC) those should be the right ones.